lumigator/python/mzai/backend/backend/api/routes/health.py

```diff
@@ -20,19 +21,19 @@ def get_health() -> HealthResponse:
 
 @router.get("/jobs/{job_id}")
 def get_job_metadata(job_id: UUID) -> JobSubmissionResponse:
-    resp = requests.get(f"{settings.RAY_DASHBOARD_URL}/api/jobs/{job_id}")
+    resp = requests.get(urllib.parse.urljoin(settings.ray_jobs, f"{job_id}"))
 
     if resp.status_code == HTTPStatus.NOT_FOUND:
         raise HTTPException(
             status_code=HTTPStatus.NOT_FOUND,
-            detail=f"Job metadata for ID: {job_id} not found",
+            detail=f"Job metadata for ID: {job_id} not found - {resp.text or ''}",
```

Comment:
🤔 I don't think we should be leaking this straight through the API. 

```suggestion
            detail=f"Job metadata for ID: {job_id} not found",
```

Reply:
I usually prefer to give the underlying info, but I have usually worked with very tightly integrated systems. I assume you'd prefer to log this info instead.

Reply:
Errors from external systems will be logged instead of sent to the user.